gst/mqtt/mqttsink.c

```diff
@@ -1067,6 +1120,92 @@ gst_mqtt_sink_set_mqtt_qos (GstMqttSink * self, const gint qos)
   self->mqtt_qos = qos;
 }
 
+/**
+ * @brief Getter for the 'ntp-sync' property.
+ */
+static gboolean
+gst_mqtt_sink_get_mqtt_ntp_sync (GstMqttSink * self)
+{
+  return self->mqtt_ntp_sync;
+}
+
+/**
+ * @brief Setter for the 'ntp-sync' property
+ */
+static void
+gst_mqtt_sink_set_mqtt_ntp_sync (GstMqttSink * self, const gboolean flag)
+{
+  self->mqtt_ntp_sync = flag;
+}
+
+/**
+ * @brief Getter for the 'ntp-srvs' property.
+ */
+static gchar *
+gst_mqtt_sink_get_mqtt_ntp_srvs (GstMqttSink * self)
+{
+  return self->mqtt_ntp_srvs;
+}
+
+/**
+ * @brief Setter for the 'ntp-srvs' property
+ */
+static void
+gst_mqtt_sink_set_mqtt_ntp_srvs (GstMqttSink * self, const gchar * pairs)
+{
+  gchar **pair_arrs = NULL;
+  guint hnum = 0;
+  gchar *pair;
+  guint i;
+
+  if (g_strcmp0 (self->mqtt_ntp_srvs, pairs) == 0)
+    return;
+
+  self->mqtt_ntp_srvs = g_strdup (pairs);
+
+  pair_arrs = g_strsplit (pairs, ",", -1);
+  if (pair_arrs == NULL)
+    return;
+
+  hnum = g_strv_length (pair_arrs);
+  if (hnum == 0)
+    goto err_free_pair_arrs;
+
+  self->mqtt_ntp_hnames =
+      g_try_malloc0 (hnum * sizeof (*self->mqtt_ntp_hnames));
+  if (!self->mqtt_ntp_hnames)
+    goto err_free_pair_arrs;
+
+  self->mqtt_ntp_ports = g_try_malloc0 (hnum * sizeof (self->mqtt_ntp_ports));
+  if (!self->mqtt_ntp_ports)
+    goto err_free_mqtt_ntp_hnames;
+
+  self->mqtt_ntp_num_srvs = hnum;
+  for (i = 0; i < hnum; i++) {
+    gchar hname[MAX_LEN_NTP_HNAME];
+    guint16 port;
+
+    pair = pair_arrs[i];
+    memset (hname, '\0', MAX_LEN_NTP_HNAME);
+
+    sscanf (pair, "%[^:]:%hd", hname, &port);
+    self->mqtt_ntp_hnames[i] = g_strdup (hname);
+    self->mqtt_ntp_ports[i] = port;
+  }
+
```

Comment:
@gichan-jang Well, in my intention, ```ntp-srvs``` is an optional property. How about explicitly declaring it in the property description?

Reply:
I asked because I thought specifying the ntp server is using ntp synchronization.
This is already well explained in the property description.
`valid only if ntp-sync is true`
If it's your intention, it's fine.

Reply:
@gichan-jang Thanks!